### ISSUE TITLE
Problems with SmarTrack&Dronie Mode.

### ISSUE THREAD

**Issue by robeitor** [wladimir-computin/FimiX8-SDFG#1]
Hi Wladimir! 

Thanks for your job! 

I recently buyed a X8 SE, and first i updated it.

In the first fly, i tried to launch the mode SMART TRACK and dont work, select the target in the app, i select the target with the rectangle and the app dont lock the target.

You can see the problem in these videos:

https://youtu.be/2ZlYNMW7wBw

https://youtu.be/58NZgAFpwu4

I use a Xiaomi A1, and I tested with other two phones, a Xiaomi A2 and Iphone 6, i could not select target with any of them.

I think that the problem are the update, but I just reinstall all firmware with your tutorial and the SmartTrack still dont working. 
I think that to downgrade with a older firmware and see that happens. 

The question are, How I Can dowload a older firmware version with your tools?

thanks you very much.

**Comment by wladimir-computin** [comment]
Hi,

currently the tool can download only the newest firmware versions automatically.
Flashing older versions requires some manual work, but it's still pretty easy.
You need to download the old firmware files, I have some of them in my other repo: https://github.com/wladimir-computin/FimiX8-RE/tree/master/firmware/X8SE

I will push an update with a detailed manual, but maybe this can help you in the meantime:

The name of the firmware files gives us only a timestamp, it does not match to the name which is shown in the app.
If the name starts with app1557384869080 for example, it means the update was released on 09.05.2019 - 08:54:29. (divide by 1000, convert from unix time).

To actually flash an old firmware, you should first generate a folder with the current firmware:
`java -jar X8UpdateGenerator-1.0.jar -i auto -f firmware-test`

Then you delete all files except the JSON file, copy the firmware files from my repo (you have to remember the firmware type, like which file is for camera, which is RC etc. Then you have to change the sysName attributes in the JSON to match the newly copied firmware files.

After that, you can generate your firmware by selecting the components you need.
Like `java -jar X8UpdateGenerator-1.0.jar -f firmware-test -u gimbal fc fc_relay`

**Comment by robeitor** [comment]
Hi,

Thanks for your help!

Ok, the only doubt is, instead to rename all SysName on JSON file,  rename the firmware files to match with JSON Sysnames ?

Do you says on your manual, that some firmwares dont are compatibles with older versions or reverse, no?  If I brick my drone, with  upgrade again with newest firmwares must be repaired, right?

Thanks!

**Comment by robeitor** [comment]
Hi,

I just do the downgrade with your instructions and it works( with renaming firmware files instead the SysName on JSON file)

But my problem with SMART TRACK still happening....  I´m waiting a solution by fimi support....

Thanks!

**Comment by clefler** [comment]
Hi, really nice work.  I'm kindof following it.  I have an error keeping me from flying (Motors Locked) and want to try an older firmware version.  Can I use the repo you have linked above, is it for the X8SE?

**Comment by maicardi** [comment]
> I recently buyed a X8 SE, and first i updated it.
> 
> In the first fly, i tried to launch the mode SMART TRACK and dont work, select the target in the app, i select the target with the rectangle and the app dont lock the target.

I have the same exact issue, but a little more info: when I have bought the FIMI X8 drone, I flew it some times before it forced me to upgrade che firmware.

The first time I flew it the smart track was working perfectly.

After the upgrade (I do not know if these facts are related) smart track do NOT work anymore: I draw the blue rectangle, the FIMI Navi app freezes for a second then the blue box disappear and I can't use the smart track mode.

> You can see the problem in these videos: 
> https://youtu.be/2ZlYNMW7wBw
> https://youtu.be/58NZgAFpwu4

Exactly the same behaviour in my case.

> I use a Xiaomi A1, and I tested with other two phones, a Xiaomi A2 and Iphone 6, i could not select target with any of them.

I have tested different phones:
- Xiaomi 9T
- Redmi Note 9
- old OnePlus One
Nothing changed.

I have tested with many FIMI Navi app versions:
1.0.17, 1.0.14, 1.0.13, 1.0.3.
Nothing changed.

The first 2 times I have tried to upgrade via the FIMI Navi app it failed, the 3rd time it succeeded.

Have you succeeded in solving the issue?

Thanks in advance.

Marco

**Comment by maicardi** [comment]
> Those modes, along with Panorama aren't supported yet...

This is NOT correct: the first times I flew the drone the Smart Track modes were working. It stopped working after the firmware upgrade, even if I am not sure the 2 things are related.

**Comment by PikoMeter** [comment]
Maybe you could try downgrading the phone app as well? 
I found a site where archives are kept.
Just a random idea
https://apkpure.com/fimi-navi/com.fimi.android.app/versions

**Comment by PikoMeter** [comment]
Hi guys, 
can anyone help me create earlier firmware bin files? I cant get the app to recognise my renamed files and write them to the bin.
They are not the same filetypes as the downloaded firmware from the backend api.
Thanks in advance
